src/server/set_family.cc

```diff
@@ -74,58 +74,74 @@ intset* IntsetAddSafe(string_view val, intset* is, bool* success, bool* added) {
   return is;
 }
 
-pair<unsigned, bool> RemoveStrSet(uint32_t now_sec, facade::ArgRange vals, CompactObj* set) {
-  unsigned removed = 0;
-  bool isempty = false;
-  DCHECK(IsDenseEncoding(*set));
+struct StringSetWrapper {
+  StringSetWrapper(const CompactObj& obj, const DbContext& db_cntx)
+      : StringSetWrapper(obj.RObjPtr(), db_cntx.time_now_ms) {
+    DCHECK(IsDenseEncoding(obj));
+  }
 
-  if (true) {
-    StringSet* ss = ((StringSet*)set->RObjPtr());
-    ss->set_time(now_sec);
+  StringSetWrapper(const SetType& st, const DbContext& db_cntx)
+      : StringSetWrapper(st.first, db_cntx.time_now_ms) {
+    DCHECK_EQ(st.second, kEncodingStrMap2);
+  }
 
-    for (string_view member : vals) {
-      removed += ss->Erase(member);
-    }
+  static void Init(CompactObj* obj) {
+    obj->InitRobj(OBJ_SET, kEncodingStrMap2, CompactObj::AllocateMR<StringSet>());
+  }
 
-    isempty = ss->Empty();
+  unsigned Add(const NewEntries& entries, uint32_t ttl_sec) const {
+    unsigned res = 0;
+    for (string_view member : EntriesRange(entries))
+      res += ss->Add(member, ttl_sec);
+    return res;
   }
 
-  return make_pair(removed, isempty);
-}
+  pair<unsigned, bool> Remove(const facade::ArgRange& entries) const {
+    unsigned removed = 0;
+    for (string_view member : entries)
+      removed += ss->Erase(member);
+    return {removed, ss->Empty()};
+  }
 
-unsigned AddStrSet(const DbContext& db_context, const NewEntries& vals, uint32_t ttl_sec,
-                   CompactObj* dest) {
-  unsigned res = 0;
-  DCHECK(IsDenseEncoding(*dest));
+  uint64_t Scan(uint64_t curs, const ScanOpts& scan_op, StringVec* res) const {
+    uint32_t count = scan_op.limit;
+    long maxiterations = count * 10;
 
-  if (true) {
-    StringSet* ss = (StringSet*)dest->RObjPtr();
-    uint32_t time_now = MemberTimeSeconds(db_context.time_now_ms);
+    do {
+      auto scan_callback = [&](const sds ptr) {
+        if (string_view str{ptr, sdslen(ptr)}; scan_op.Matches(str))
+          res->emplace_back(str);
+      };
+      curs = ss->Scan(curs, scan_callback);
+    } while (curs && maxiterations-- && res->size() < count);
+    return curs;
+  }
 
-    ss->set_time(time_now);
+  StringSet* operator->() const {
+    return ss;
+  }
 
-    for (auto member : EntriesRange(vals)) {
-      res += ss->Add(member, ttl_sec);
-    }
+  auto Range() const {
+    auto transform = [](const sds ptr) { return string_view{ptr, sdslen(ptr)}; };
+    return base::it::Transform(transform, base::it::Range(ss->begin(), ss->end()));
   }
 
-  return res;
-}
+ private:
+  StringSetWrapper(void* robj_ptr, uint64_t now_ms) : ss(static_cast<StringSet*>(robj_ptr)) {
+    ss->set_time(MemberTimeSeconds(now_ms));
```

Comment:
Just make sure that we don't call this on paths that don't require this. I went over the cases and they seemed fine but just double checking

Reply:
I checked too. And if a command path existed, it was likely missing, not abundant 🤷🏻‍♂️